### PR TITLE
Tweak outreach map

### LINKED ISSUE #284: Outreach map renovation
Get the outreach map up to date, and furnish it a bit more. Make it more functional.

### PR DESCRIPTION
**Pull request by PsyCommando**
## Description of changes
Remake the outreach map to be a bit more up to date and interesting.

## Changelog
:cl:
tweak: Tweaked a lot of things.
/:cl:

### CODE REVIEW

**Comment by PsyCommando** [comment]
Gotta clean the trash in this PR

**Comment by PsyCommando** [comment]
Alright, so it passes CI now.

File: maps/kleibkhar/kleibkhar_exoplanet.dm [hunk 1]
```diff
@@ -1,51 +1,87 @@
...
-	color = "#407c40"
-	grass_color = "#407c40"
-	planetary_area = /area/exoplanet/kleibkhar
-	rock_colors = list(COLOR_ASTEROID_ROCK, COLOR_GRAY80, COLOR_BROWN)
-	plant_colors = list("#215a00","#195a47","#5a7467","#9eab88","#6e7248", "RANDOM")
-	surface_color = COLOR_DARK_GREEN_GRAY
-	water_color = COLOR_BLUE_GRAY
-	crust_strata = /decl/strata/base_planet
+/datum/map/kleibkhar/build_exoplanets()
+	log_debug("MAP: building exoplanets!") //#REMOVEME
```
Comment (NataKilar) [inline maps/kleibkhar/kleibkhar_exoplanet.dm]: Debug print

File: maps/kleibkhar/kleibkhar_exoplanet.dm [hunk 2]
```diff
@@ -99,21 +135,73 @@
...
+	floor_type = /turf/exterior/barren/mining
+	mineral_turf = /turf/exterior/wall/random/kleibkhar
+
+//////////////////////////////////////////////////////////////////////////
+// Strata
+//////////////////////////////////////////////////////////////////////////
+/decl/strata/kleibkhar
+	default_strata_candidate = TRUE
+
+/decl/strata/kleibkhar/subterrane
```
Comment (NataKilar) [inline maps/kleibkhar/kleibkhar_exoplanet.dm]: Subterranean or subterrane? I know both are words.

File: maps/outreach/outreach_defines.dm [hunk 3]
```diff
@@ -1,33 +1,31 @@
...
 
 	lobby_screens = list(
 		'maps/outreach/lobby/exoplanet.png'
 	)
+	//Lobby tracks moved to outreach_music.dm otherwise would cause all sort of problems
 
-	starting_money = 5000
-	department_money = 0
-	salary_modifier = 0.2
+	//#REMOVEME: Shouldn't get commited!
```
Comment (NataKilar) [inline maps/outreach/outreach_defines.dm]: Should this be removed?

File: maps/outreach/outreach_exoplanet.dm [hunk 4]
```diff
@@ -1,69 +1,224 @@
...
+	water_material        = /decl/material/gas/chlorine
+	ice_material          = /decl/material/gas/chlorine
 
-	rock_colors = list(COLOR_GRAY80, COLOR_PALE_GREEN_GRAY, COLOR_PALE_BTL_GREEN)
-	plant_colors = list(COLOR_PALE_PINK, COLOR_PALE_GREEN_GRAY, COLOR_CIVIE_GREEN)
-	surface_color = COLOR_PALE_GREEN_GRAY
-	water_color = COLOR_BOTTLE_GREEN
-	crust_strata = /decl/strata/sedimentary
+	//Sector Access
+	free_landing          = FALSE
```
Comment (NataKilar) [inline maps/outreach/outreach_exoplanet.dm]: This needs to be TRUE, otherwise nothing can land and take off from the surface.

File: maps/outreach/outreach_overmap.dm [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	name = "Castor"
+
+/////////////////////////////////////////////////////////////////////////////////
+// Overmap Details
+/////////////////////////////////////////////////////////////////////////////////
+/datum/overmap/outreach
+	event_areas = 10
+	map_size_x  = 50
+	map_size_y  = 50
+	var/map_file// = "maps/outreach/outreach-overmap.dmm"
```
Comment (NataKilar) [inline maps/outreach/outreach_overmap.dm]: Should this be commented out?

File: maps/outreach/outreach_overmap.dm [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+		return
+
+	//Otherwise just spawn the star
+	var/turf/overmap_center = locate(round(map_size_x/2), round(map_size_y/2), assigned_z)
+	new /obj/effect/overmap/star/outreach(overmap_center)
+
+/////////////////////////////////////////////////////////////////////////////////
+// Supply Shuttle Handling
+/////////////////////////////////////////////////////////////////////////////////
+#if 0
```
Comment (NataKilar) [inline maps/outreach/outreach_overmap.dm]: Make this an actual define or remove it

File: maps/outreach/outreach_weather.dm [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+/decl/state/weather/rain/storm/outreach/handle_exposure_effects(mob/living/M, obj/abstract/weather_system/weather)
+	if(prob(80))
+		to_chat(M, SPAN_DANGER("You were doused with acid rain!"))
+		var/datum/reagents/R = new(10, global.temp_reagents_holder)
+		R.splash(M, rand(1, R.total_volume), min_spill = 0, max_spill = 0)
+
+/decl/state/weather/salt_flurry
+	name       = "Salt Flurry"
+	icon_state = "snowfall_light"
+	descriptor = "The wind is blowing Chlorine salts everywhere."
```
Comment (NataKilar) [inline maps/outreach/outreach_weather.dm]: Chlorine -> chlorine

File: maps/outreach/turfs/outreach_turf_flooded.dm [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+///////////////////////////////////////////////////////////////////////////////////
+/turf/simulated/floor/asteroid/outreach/water/chlorine
+	color          = "#d2e0b7"
+	open_turf_type = /turf/simulated/floor/asteroid/outreach
+	prev_type      = /turf/simulated/floor/asteroid/outreach
+
+/turf/simulated/floor/asteroid/outreach/water/chlorine/Initialize(ml, floortype)
+	. = ..()
+	make_flooded(TRUE)
+	add_fluid(/decl/material/gas/chlorine, FLUID_MAX_DEPTH)
```
Comment (NataKilar) [inline maps/outreach/turfs/outreach_turf_flooded.dm]: Shouldn't this be hydrochloric acid? Liquid chlorine would rapidly boil

File: mods/persistence/modules/overmap/_overmap.dm [hunk 9]
```diff
@@ -1,30 +1,30 @@
-// Complete override since Persistence hotloads it maps, and most of these functions must be delayed.
```
Comment (NataKilar) [inline mods/persistence/modules/overmap/_overmap.dm]: If this has been depreciated you can delete the file

File: mods/persistence/modules/overmap/exoplanets/_exoplanet.dm [hunk 10]
```diff
@@ -1,8 +1,10 @@
+//#TODO: Gotta fix this so we just run the exoplanet's gen code on startup instead to properly update areas and stuff for unsaved turfs.
```
Comment (NataKilar) [inline mods/persistence/modules/overmap/exoplanets/_exoplanet.dm]: Has this been fixed?

File: code/__defines/overmap.dm [hunk 11]
```diff
@@ -29,3 +29,11 @@
 #define SENSOR_COEFFICENT 1000
 
 #define OVERMAP_ID_SPACE "Deep Space"
+
+#define ZLEVEL_STATION    BITFLAG(0)
+#define ZLEVEL_ADMIN      BITFLAG(1)
+#define ZLEVEL_CONTACT    BITFLAG(2)
+#define ZLEVEL_PLAYER     BITFLAG(3)
+#define ZLEVEL_SEALED     BITFLAG(4)
+#define ZLEVEL_PERSISTENT BITFLAG(5) //If set the z-level will be added to the saved level list
```
Comment (NataKilar) [inline code/__defines/overmap.dm]: The last two defines should be moved into the Persistence modpack

File: code/game/machinery/doors/airlock_control.dm [hunk 12]
```diff
@@ -126,22 +126,7 @@
...
 /obj/machinery/airlock_sensor/buildable
-	uncreated_component_parts = list(
-		/obj/item/stock_parts/power/apc = 1
-	)
-
-/obj/machinery/airlock_sensor/Initialize(mapload, d, populate_parts)
-	. = ..()
-	update_icon()
-
-/obj/machinery/airlock_sensor/examine(mob/user, distance, infix, suffix)
```
Comment (NataKilar) [inline code/game/machinery/doors/airlock_control.dm]: Was this examine text replaced somewhere?

File: code/game/objects/effects/effect_system.dm [hunk 13]
```diff
@@ -69,7 +69,8 @@ steam.start() -- spawns the effect
 /datum/effect/effect/system/steam_spread/start()
 	var/i = 0
 	for(i=0, i<src.number, i++)
-		addtimer(CALLBACK(src, /datum/effect/effect/system/proc/spread, i), 0)
+		spread(i)
+		//addtimer(CALLBACK(src, /datum/effect/effect/system/proc/spread, i), 0)
```
Comment (NataKilar) [inline code/game/objects/effects/effect_system.dm]: Why was this changed? I believe the 0 delay callback is intentional.

File: code/modules/organs/internal/lungs.dm [hunk 14]
```diff
@@ -80,6 +80,8 @@
 		poison_types =        species.poison_types || list(/decl/material/gas/chlorine = TRUE)
 		exhale_type =         species.exhale_type  || /decl/material/gas/carbon_dioxide
 	else
+		//Species-less lungs should be illegal
+		log_debug("Got null species")
```
Comment (NataKilar) [inline code/modules/organs/internal/lungs.dm]: Might want to make this debug message more specific

File: code/modules/overmap/sectors.dm [hunk 15]
```diff
@@ -60,6 +60,7 @@ var/global/list/known_overmap_sectors
 	. = ..()
 
 /obj/effect/overmap/visitable/Initialize()
+	log_debug("[src]([x],[y],[z]) creating sector.") //#REMOVEME
```
Comment (NataKilar) [inline code/modules/overmap/sectors.dm]: Left over debug lines?

File: code/modules/overmap/sectors.dm [hunk 16]
```diff
@@ -125,6 +126,7 @@ var/global/list/known_overmap_sectors
 	return get_filtered_areas(list(/proc/area_belongs_to_zlevels = map_z))
 
 /obj/effect/overmap/visitable/proc/find_z_levels()
+	log_debug("[src]([x],[y],[z]) finding z levels. Current z is [z].") //#REMOVEME
```
Comment (NataKilar) [inline code/modules/overmap/sectors.dm]: Ditto

File: code/modules/overmap/ships/landable.dm [hunk 17]
```diff
@@ -116,12 +116,14 @@
 		landmark.shuttle_name = shuttle
 		LAZYDISTINCTADD(initial_generic_waypoints, landmark.landmark_tag) // this is us being user-friendly: it means we register it properly regardless of whether the mapper put the tag in initial_restricted_waypoints
 
+	/* //#OUTREACH: Since we delete those right when creating them, might as well just comment this out
```
Comment (NataKilar) [inline code/modules/overmap/ships/landable.dm:117]: This is fine, but you might as well delete the part of the code that actually deletes them as well.

**Review by NataKilar — CHANGES_REQUESTED** [review]

File: maps/kleibkhar/kleibkhar_exoplanet.dm (hunk 2 above)
Comment (PsyCommando) [reply maps/kleibkhar/kleibkhar_exoplanet.dm]: Eh?

File: maps/outreach/outreach_defines.dm (hunk 3 above)
Comment (PsyCommando) [reply maps/outreach/outreach_defines.dm]: Oh, yep.

File: maps/outreach/outreach_exoplanet.dm (hunk 4 above)
Comment (PsyCommando) [reply maps/outreach/outreach_exoplanet.dm]: Speaking of that. I really don't think we should let shuttles land anywhere on the planet, but on docking areas?

Because the shuttle code still lets you land on any z-levels of the planet, and pretty much anywhere without an area protection thing. 
That in turn is going to mess with the base turfs when the thing takes off. And I just can't guarantee it won't cause problems or end up being used for griefing.

For reference, landing on a transition edge is going to permanently fuck the map. The underground turfs ended up being less trouble being simulated turfs for allowing laying down cable and etc,. So a shuttle landing there is very bad..

I can turn it back on, but like, can we at least add support for docking ports, so we don't have to deal with shuttle damage again? It's really hard to recover anything that a shuttle destroys.

File: maps/outreach/outreach_overmap.dm (hunk 5 above)
Comment (PsyCommando) [reply maps/outreach/outreach_overmap.dm]: Oh, that actually reminds me I didn't test the overmap -_-
But yeah that's fine. We really shouldn't need to pre-generate the overmap.

File: mods/persistence/modules/overmap/exoplanets/_exoplanet.dm (hunk 10 above)
Comment (PsyCommando) [reply mods/persistence/modules/overmap/exoplanets/_exoplanet.dm]: Nope. I just added a todo for it.

Comment (PsyCommando) [reply mods/persistence/modules/overmap/exoplanets/_exoplanet.dm]: Nope. This will require work upstream. I think loaf mentioned he'd fix planet so they init at the same time as everything else. But it probably will take a while.

File: code/__defines/overmap.dm (hunk 11 above)
Comment (PsyCommando) [reply code/__defines/overmap.dm]: Can't do that. It's used in the map. It would fail unit test on the map if the modpack wasn't loaded.
Those probably should get moved upstream too. I was going to when getting the lateral z stuff upstream, but that's gonna be a while off..

File: code/game/objects/effects/effect_system.dm (hunk 13 above)
Comment (PsyCommando) [reply code/game/objects/effects/effect_system.dm]: It was removed because it caused massive lag when you had a few geysers on the map, and a lot of timer runtimes. Plus, spread already is async anyways?
I actually rewrote that effect for geysers using particles though.

File: code/modules/organs/internal/lungs.dm (hunk 14 above)
Comment (PsyCommando) [reply code/modules/organs/internal/lungs.dm]: Yeah, that shouldn't be in here. I was debugging why mobs would runtime when breathing.

File: mods/persistence/modules/overmap/_overmap.dm (hunk 9 above)
Comment (PsyCommando) [reply mods/persistence/modules/overmap/_overmap.dm]: kk

File: code/game/machinery/doors/airlock_control.dm (hunk 12 above)
Comment (PsyCommando) [reply code/game/machinery/doors/airlock_control.dm]: Probably something that slipped during the merge..

File: code/game/objects/effects/effect_system.dm (hunk 13 above)
Comment (PsyCommando) [reply code/game/objects/effects/effect_system.dm]: I'll revert that

File: code/modules/overmap/ships/landable.dm (hunk 17 above)
Comment (PsyCommando) [reply code/modules/overmap/ships/landable.dm:117]: Just for reference, that was to fix the hard deletes. 
I'd suggest maybe doing that in another PR? Because this one is extremely bloated as it is.

File: code/controllers/subsystems/mapping.dm [hunk 18]
```diff
@@ -109,6 +107,10 @@ SUBSYSTEM_DEF(mapping)
 			PRINT_STACK_TRACE("Missing z-level data object for z[num2text(z)]!")
 		level.setup_level_data()
 
+	// Generate turbolifts last!!!
+	for(var/obj/abstract/turbolift_spawner/turbolift as anything in turbolifts_to_initialize)
```
Comment (NataKilar) [inline code/controllers/subsystems/mapping.dm:111]: Is this something specific for Persistence or could it go upstream eventually?

File: code/game/objects/items/stacks/dust.dm [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+	for(var/mtype in matter)
+		var/amount = matter[mtype]
+		if(amount)
+			R.add_reagent(mtype,  amount * REAGENT_UNITS_PER_MATERIAL_UNIT)
+	if(R.total_volume > 0)
+		R.splash(hit_atom, R.total_volume)
+	use(amount)
+
+// Dust override
+/decl/material
```
Comment (NataKilar) [inline code/game/objects/items/stacks/dust.dm:51]: Instinct tells me this should be in the base material file, but the Neb maints can decide once/if it goes upstream

File: code/modules/fabrication/designs/imprinter/designs_misc_circuits.dm [hunk 20]
```diff
@@ -485,6 +485,11 @@
 /datum/fabricator_recipe/imprinter/circuit/photocopier
 	path = /obj/item/stock_parts/circuitboard/photocopier
 
+/datum/fabricator_recipe/imprinter/circuit/geothermal_generator
+	path = /obj/item/stock_parts/circuitboard/geothermal
+
+/datum/fabricator_recipe/imprinter/circuit/radiocarbon_spectrometer
+	path = /obj/item/stock_parts/circuitboard/radiocarbon_spectrometer
```
Comment (NataKilar) [inline code/modules/fabrication/designs/imprinter/designs_misc_circuits.dm:492]: New line here if you get the chance

File: code/unit_tests/map_tests.dm [hunk 21]
```diff
@@ -633,6 +634,9 @@
 	if(!isStationLevel(source_turf.z))
 		return TRUE
 
+	if(locate(/obj/abstract/landmark/skip_test) in source_turf)
```
Comment (NataKilar) [inline code/unit_tests/map_tests.dm:637]: Is this the only test this should be checked?

File: maps/outreach/outreach_machinery.dm [hunk 22]
```diff
@@ -0,0 +1,219 @@
...
+////////////////////////////////////////////////////////////////////////
+// Docking Beacon
+////////////////////////////////////////////////////////////////////////
+/obj/machinery/docking_beacon/mapped
+	anchored = TRUE
+
+////////////////////////////////////////////////////////////////////////
+// Defense Turret
+////////////////////////////////////////////////////////////////////////
+/obj/machinery/porta_turret/stationary/stun
```
Comment (NataKilar) [inline maps/outreach/outreach_machinery.dm:206]: Making a note that we'll need to replace these with the new turrets once they're in

File: code/controllers/subsystems/mapping.dm (hunk 18 above)
Comment (PsyCommando) [reply code/controllers/subsystems/mapping.dm:111]: I think it went upstream already?

Comment (PsyCommando) [reply code/controllers/subsystems/mapping.dm:111]: It's not unique to outreach, any dynamically loaded maps like away_sites would have that issue if they had elevators.

File: code/unit_tests/map_tests.dm (hunk 21 above)
Comment (PsyCommando) [reply code/unit_tests/map_tests.dm:637]: For now yes. But it's a work in progress tool for convenience.

**Review by NataKilar — APPROVED** [review]
Further map adjustments can be made in future PRs. This seems fine.
